dpctl/tensor/libtensor/include/kernels/elementwise_functions/pow.hpp

```diff
@@ -0,0 +1,292 @@
+//=== POW.hpp -   Binary function POW                    ------  *-C++-*--/===//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===---------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines kernels for elementwise evaluation of POW(x1, x2)
+/// function.
+//===---------------------------------------------------------------------===//
+
+#pragma once
+#include <CL/sycl.hpp>
+#include <cstddef>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+#include "utils/offset_utils.hpp"
+#include "utils/type_dispatch.hpp"
+#include "utils/type_utils.hpp"
+
+#include "kernels/elementwise_functions/common.hpp"
+#include <pybind11/pybind11.h>
+
+namespace dpctl
+{
+namespace tensor
+{
+namespace kernels
+{
+namespace pow
+{
+
+namespace py = pybind11;
+namespace td_ns = dpctl::tensor::type_dispatch;
+namespace tu_ns = dpctl::tensor::type_utils;
+
+template <typename argT1, typename argT2, typename resT> struct PowFunctor
+{
+
+    using supports_sg_loadstore = std::negation<
+        std::disjunction<tu_ns::is_complex<argT1>, tu_ns::is_complex<argT2>>>;
+    using supports_vec =
+        std::negation<std::disjunction<tu_ns::is_complex<argT1>,
+                                       tu_ns::is_complex<argT2>,
+                                       std::is_integral<argT1>,
+                                       std::is_integral<argT2>>>;
+
+    resT operator()(argT1 in1, argT2 in2)
+    {
+        if constexpr (std::is_integral_v<argT1> || std::is_integral_v<argT2>) {
+            if constexpr (std::is_signed_v<argT2>) {
+                if (in2 < 0) {
+                    // invalid; return 0
+                    return resT(0);
```

Comment:
I think it would be better if you raised an error here as implemented in numpy
```
import numpy

numpy.power(numpy.array([2]),-2)
ValueError: Integers to negative integer powers are not allowed.
```

Reply:
Raising an error inside of a kernel is not something we've been doing inside of a kernel (parallel, some memory might already be changed, etc.)

We did not raise inside of kernels when implementing integer advanced indexing for this reason, so I think it should be left this way.